src/util/date-input-helper.ts

```diff
@@ -1,34 +1,124 @@
 import dayjs from "dayjs";
+import isBetween from "dayjs/plugin/isBetween";
+import { INVALID_VALUE } from "../date-input/stand-alone-input";
+import { ChangeValueTypes, RawInputValues } from "../date-input";
+
+dayjs.extend(isBetween);
 
 export namespace DateInputHelper {
-    export const validate = (beginDate: string, afterDate: string): boolean => {
+    export const validate = (
+        startDate: string,
+        endDate: string,
+        disabledDates?: string[] | undefined,
+        between?: string[] | undefined
+    ): boolean => {
         let pass = false;
 
-        if (!beginDate.length || !afterDate.length) return false;
+        if (startDate.length === 0 || endDate.length === 0) return false;
 
-        if (dayjs(beginDate).isAfter(afterDate)) {
-            pass = false;
+        if (dayjs(startDate).isBefore(endDate)) {
+            pass = true;
+        }
 
-            return pass;
+        if (disabledDates && disabledDates.length) {
+            pass = !disabledDates.some((value) =>
+                [startDate, endDate].includes(value)
+            );
+
+            if (!pass) return false;
         }
 
-        if (dayjs(afterDate).isBefore(beginDate)) {
+        if (between && between.length) {
+            const result = [startDate, endDate].map((selectedDate) =>
+                dayjs(selectedDate).isBetween(
+                    between[0],
+                    between[1],
+                    "day",
+                    "[]"
+                )
+            );
+
+            pass = result.every(Boolean);
+
+            if (!pass) return false;
+        }
+
+        if (dayjs(startDate).isAfter(endDate)) {
             pass = false;
 
             return pass;
         }
 
-        if (dayjs(beginDate).isBefore(afterDate)) {
-            pass = true;
+        return pass;
+    };
+
+    export const singleValidation = (
+        value: string,
+        disabledDates?: string[] | undefined,
+        between?: string[] | undefined
+    ) => {
+        let pass = true;
```

Comment:
rename this fn to `validateSingle`

no need to maintain this pass variable. can do something like

```
() => {
  if (cond1) {
    if (cond2) {
      return false
    }
  }
  return true;
}
```

Reply:
Note:

my bad, i couldn't understand clearly. i updated to this. let me know if it is not what you expected.
```
() => {
  if (value.length === 0) {
    return false;
  }

  if (disabledDates && disabledDates.some((disabledDate) => value === disabledDate)) {
    return false;
  }

  if (between && !dayjs(value).isBetween(between[0], between[1], "day", "[]")) {
    return false;
  }

  return true;
}

```